src/Workspaces/CSharp/Portable/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -1050,5 +1050,123 @@ public static T WithCommentsFrom<T>(this T node, SyntaxToken leadingToken, Synta
             => node
             .WithTrailingTrivia(node.GetTrailingTrivia().FilterComments(addElasticMarker: true))
             .WithLeadingTrivia(node.GetLeadingTrivia().FilterComments(addElasticMarker: true));
+
+        // Once https://github.com/dotnet/roslyn/issues/30116 is implemented, we can switch this API
+        // and other extension methods in this file that use this API to be language-agnostic extensions.
+        // TODO: File a tracking a track bug for this API cleanup.
+        public static ValueUsageInfo GetValueUsageInfo(this SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (node is ExpressionSyntax expression)
+            {
+                if (expression.IsRightSideOfDotOrArrow())
+                {
+                    expression = expression.Parent as ExpressionSyntax;
+                }
+
+                node = expression.WalkDownParentheses();
+            }
+
+            var operation = semanticModel.GetOperation(node, cancellationToken);
+            if (operation != null)
+            {
+                var valueUsageInfo = operation.GetValueUsageInfo(out var isInErrorContext);
+
+                // Workaround for https://github.com/dotnet/roslyn/issues/18722 (lack of IInvalidInvocationOperation).
+                // If there is an overload resolution failure for the invocation, then the operation tree
+                // has an IInvalidOperation for the invocation and argument syntax nodes, and
+                // we cannot correctly determine the RefKind of the argument from the operation tree.
+                // So we workaround this scenario by performing syntactic checks.
+                if (isInErrorContext && node is ExpressionSyntax expressionNode)
```

Comment:
ick....

I'd prefer to wait on 18722 being addressed.  Otherwise, we have two entirely different codepaths to ask the same question.  Ensuring they're in sync and don't have problems is not something i'd prefer us to do.

Reply:
I agree that this is not an ideal workaround, but note that the remaining syntactic checks are purely around ref/out/in arguments, which is still not large amount of code in error path. We will discuss costing around `IInvalidInvocationOperation` next week, and I am going to push for us addressing it soon. If we end up identifying that this is too costly to fit into preview2 or dev16 timeframe, I will discuss with @jinujoseph on the best path ahead here.